packages/workbox-strategies/src/NetworkOnly.ts

```diff
@@ -42,15 +68,29 @@ class NetworkOnly extends Strategy {
       assert!.isInstance(request, Request, {
         moduleName: 'workbox-strategies',
         className: this.constructor.name,
-        funcName: 'handle',
+        funcName: '_handle',
         paramName: 'request',
       });
     }
 
-    let error;
-    let response;
+    let error: Error | undefined = undefined;
+    let response: Response | undefined;
+
     try {
-      response = await handler.fetch(request);
+      const promises: Promise<Response|undefined>[] = [handler.fetch(request)];
+
+      if (this._networkTimeoutSeconds) {
+        const timeoutPromise = new Promise((resolve: (_: undefined) => void) => {
+          setTimeout(resolve, this._networkTimeoutSeconds * 1000);
+        });
+        promises.push(timeoutPromise);
+      }
+
+      response = await Promise.race(promises);
```

Comment:
I know `Promise.race()` is often not the right choice... but here I think it is?